placement/placement.go

```diff
@@ -75,13 +75,16 @@ type StoragePlacement interface {
 
 	// JoinCluster adds a new cluster to an existing database and rebalances
 	// shards onto that cluster
-	JoinCluster(db string, c schema.ClusterProperties) error
+	JoinCluster(db string, c schema.ClusterProperties, config []byte) error
```

Comment:
Would this be slightly more in line with our KV interfaces if `config` was a `proto.Message` instead that we `Marshal` inside this method?